src/controller/CHIPCluster.cpp

```diff
@@ -42,10 +42,11 @@ CHIP_ERROR ClusterBase::Associate(DeviceProxy * device, EndpointId endpoint)
     return err;
 }
 
+/**
+ * To be used in a test context where a SecureSession is already present
+ */
 CHIP_ERROR ClusterBase::AssociateWithGroup(DeviceProxy * device, GroupId groupId)
```

Comment:
It seems like the right fix should be to change AssociateWithGroup to take the args relevant to a group and not take a DeviceProxy at all....  We should not be creating separate APIs/codepaths for tests vs non-tests if we can avoid it.